GrocifyApp.UI/src/components/Register.jsx

```diff
@@ -0,0 +1,163 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { ReactSVG } from 'react-svg';
+
+//Internal components
+import CustomInput from './CustomInput';
+import UserPassword from './UserPassword';
+
+//Assets & Css
+import ReactLogo from '../assets/logo_with_text.svg';
+import ArrowIcon from '../assets/arrow-ic.svg';
+import ApiEndpoints from '../consts/ApiEndpoints';
+import stylesAuth from './Auth.module.scss';
+import styles from './Register.module.scss';
+
+//Consts
+import { GenericConsts, AuthConsts, ButtonConsts } from '../consts/ENConsts';
+import AppRoutes from '../consts/AppRoutes';
+import { useEffect } from 'react';
+
+const RegisterForm = () => {
+    const navigate = useNavigate();
+    const [name, setName] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [confirmPassword, setConfirmPassword] = useState('');
+    const [showPart1, setShowPart1] = useState(true);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [isButtonDisabled, setButtonDisabled] = useState(true);
+    
+    useEffect(() => {
```

Comment:
if name is incorrect or email is incorrect disable button and setErrorMessage. else button enabled.

you can remove handleProceed because you only enable button if the user can proceed